apps/plasmo/src/components/Calendar/helpers/dateUtils.ts

```diff
@@ -1,5 +1,33 @@
 import type { ProcessedDay } from "../types/types"
 
+// Mapping of Nepali months to approximate Gregorian (English) dates
+const nepaliToEnglishMonthMap: {
+  [key: string]: { month: number; startDay: number; year: number }
+} = {
+  "01": { month: 3, startDay: 14, year: 0 }, // Baisakh starts ~ April 14
+  "02": { month: 4, startDay: 14, year: 0 }, // Jestha starts ~ May 14
+  "03": { month: 5, startDay: 15, year: 0 }, // Ashadh starts ~ June 15
+  "04": { month: 6, startDay: 16, year: 0 }, // Shrawan starts ~ July 16
+  "05": { month: 7, startDay: 17, year: 0 }, // Bhadra starts ~ August 17
+  "06": { month: 8, startDay: 17, year: 0 }, // Ashwin starts ~ September 17
+  "07": { month: 9, startDay: 17, year: 0 }, // Kartik starts ~ October 17
+  "08": { month: 10, startDay: 16, year: 0 }, // Mangsir starts ~ November 16
+  "09": { month: 11, startDay: 16, year: 0 }, // Poush starts ~ December 16
+  "10": { month: 0, startDay: 14, year: 1 }, // Magh starts ~ January 14
+  "11": { month: 1, startDay: 13, year: 1 }, // Falgun starts ~ February 13
+  "12": { month: 2, startDay: 14, year: 1 } // Chaitra starts ~ March 14
+}
+
+// Convert Nepali year to English year
+const getEnglishYear = (nepaliYear: number, nepaliMonth: string): number => {
+  // Base conversion (approximate)
+  const baseEnglishYear = nepaliYear - 57
+
+  // Adjust for months that span into the next year
+  const monthData = nepaliToEnglishMonthMap[nepaliMonth]
+  return baseEnglishYear + (monthData?.year || 0)
+}
+
```

Comment:
@pranishpaudel  please use `nepali-datetime` npm package, the English month change date is not even valid for each year, it varies.